static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -238,6 +246,49 @@ function SelectProjectStep({
   );
 }
 
+function usePublicDSN({
+  organization,
+  project,
+}: {
+  organization: Organization | null | undefined;
+  project: Project | null | undefined;
+}) {
+  const api = useApi();
+  const [response, setResponse] = useState<
+    RequestState<string | null> & {loading: boolean}
+  >({
+    type: 'initial',
+    loading: false,
+  });
+  useEffect(() => {
+    if (!organization || !project) {
+      return;
+    }
+    setResponse(v => ({...v, type: 'initial', data: null, loading: true}));
```

Comment:
You'll be able to just use `{type: 'loading'}` as the loading state instead of creating a new intersection type.